### PR TITLE
Fix: medication X button passes wrong ID and removed flawed archival mechanism that was only used to archive re-staged medications after re-editing

### PR DESCRIPTION
**Pull request by LiamStanziani**
## Summary
  Fixes two interdependent bugs in the prescription staging UI where the X button on drug cards failed to remove drugs from the server stash, and a hidden flag caused unintended permanent archival of medications from the database.

> **Note:** These bug numbers are based on a google document listing medication issues found, not other issues tracked in GitHub.

Original PR in openo-beta: https://github.com/openo-beta/Open-O/pull/2343 

  ## Problem
  **Bug `#5` (Regression):** `deletePrescribingDrugFromUI()` passed `drugId`
  (DrugReferenceId) to `deletePrescribe()` instead of `uiRefId` (randomId).
  The server looks up stash items by randomId, so the lookup always failed silently
  — the X button removed the drug from the UI but left it in the server stash.

  **Bug `#6` (Pre-existing, masked by Bug `#5`):** The `deleteOnCloseRxBox` hidden
  field was set to `'true'` when "Edit Rx" was clicked but never reset. This caused
  `deletePrescribe()` to also call `deleteRxOnCloseRxBox()`, which permanently
  archived drugs from the database without user confirmation. Because Bug `#5`
  prevented `deletePrescribe()` from reaching that code path, this bug was masked.

  ## Solution
  **Bug `#5`:** Pass `uiRefId` (the randomId) instead of `drugId` to
  `deletePrescribe()`. Remove the now-unused `drugId` parameter from
  `deletePrescribingDrugFromUI()`.

**Bug `#6`:** Remove the entire `deleteOnCloseRxBox` mechanism — a flawed 2010-era
  feature (commit 75940c0, SourceForge Bug: https://sourceforge.net/p/oscarmcmaster/bugs/1336/) that silently changed the X button from
  "remove from staging" to "permanently archive from database" based on hidden
  state, with no visual indication or confirmation. This is unsafe behavior for a
  healthcare system. Removed: the hidden input, flag setter, conditional check in
  `deletePrescribe()`, the `deleteRxOnCloseRxBox()` function, and the
  server-side `DeleteRxOnCloseRxBox()` method in `RxDeleteRx2Action.java`.

**Bug `#7`:** Removed the underlying functionality, this is no longer applicable, resolved in this PR

  ## Additional improvements
  - **CSRF protection:** `deletePrescribe()` changed from GET to POST with CSRF
    token header, so it is now covered by CSRFGuard's protected methods
    (`POST,PUT,DELETE`)
  - **Prototype.js removal:** Replaced `Ajax.Request` with `fetch` and
    Prototype `$()` calls with `document.getElementById()`
  - **jQuery removal:** Replaced `jQuery("#...").remove()` with vanilla
    `document.getElementById()?.remove()`
  - **Error handling:** Added error logging for failed stash removal requests
  - **JSDoc fix:** Corrected misleading `@param` documentation on
    `uncheckReRxForExistingPrescribedDrug()`
- **Additional javadoc fixes and additions**

## Summary by Sourcery

Fix prescription staging drug removal to use the correct UI reference ID and remove the legacy delete-on-close mechanism that could permanently archive medications, while modernizing the related client-server interaction.

Bug Fixes:
- Ensure the X button correctly removes staged drugs from the server stash by passing the UI randomId instead of the drug database ID.
- Eliminate the hidden delete-on-close flag and its server-side handler that could silently archive medications when editing prescriptions.

Enhancements:
- Convert the stash deletion request to a POST with CSRF protection and replace Prototype/jQuery AJAX utilities with fetch and vanilla DOM APIs.
- Add error logging for failed stash removal requests and improve JSDoc/Javadoc documentation and parameter descriptions.

## Summary by Sourcery

Fix prescription staging drug removal to target the correct UI reference ID and remove the legacy delete-on-close archival mechanism, while modernizing the client and server handling of stash deletions.

Bug Fixes:
- Ensure drug removal from the staging area uses the UI randomId so the server correctly deletes items from the prescription stash.
- Eliminate the hidden delete-on-close mechanism that could trigger unintended archival of prescriptions when editing from the preview dialog.

Enhancements:
- Convert the prescription stash deletion endpoint to a POST request with CSRF token support and structured error logging.
- Replace Prototype and jQuery usages in the prescription staging UI with fetch and vanilla DOM APIs for AJAX and element removal.

Documentation:
- Add and update JSDoc and Javadoc comments for the prescription search/staging UI and deletion action, including clearer parameter and behavior descriptions.

### CODE REVIEW

File: src/main/webapp/oscarRx/SearchDrug3.jsp [hunk 1]
```diff
@@ -1723,47 +1743,26 @@
...
+    }).then(function (response) {
+      if (!response.ok) {
+        console.error('Failed to remove drug from stash (HTTP ' + response.status + ')');
       }
+      document.getElementById("set_" + randomId)?.remove();
+      document.getElementById("prescriptionMoreLessLink_" + randomId)?.remove();
+      document.getElementById("deleteMedicationFromPrescription_" + randomId)?.remove();
+    }).catch(function (error) {
+      console.error('deletePrescribe error:', error);
     });
```
[the remark this answers is not in the document]
Comment (LiamStanziani) [reply src/main/webapp/oscarRx/SearchDrug3.jsp:1763]: Already addressed in the original PR, not an issue
